### PR TITLE
universal-android-debloater: init at 0.6.1

### LINKED ISSUE #199619: [package request] Universal Android Debloater
**Project description**
Cross-platform GUI written in Rust using ADB to debloat non-rooted android devices. Improve your privacy, the security and battery life of your device. 

**Metadata**

* homepage URL: https://github.com/0x192/universal-android-debloater
* source URL: https://github.com/0x192/universal-android-debloater
* license: GPL-3.0
* platforms: linux, darwin

### LINKED ISSUE #251561: Package request: Universal Android Debloater GUI
**Project description**

Cross-platform GUI written in Rust using ADB to debloat non-rooted android devices. Improve your privacy, the security and battery life of your device. 

**Metadata**

* homepage URL:
* source URL: https://github.com/0x192/universal-android-debloater
* license: gpl-3
* platforms: linux, macos, windows

### PR DESCRIPTION
**Pull request by sugar700**
## Description of changes

Based on https://github.com/NixOS/nixpkgs/pull/267729. Fixes #199619. Fixes #251561.

cc @gesperon

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by xfix** [comment]
Seems like the community has moved over to https://github.com/Universal-Debloater-Alliance/universal-android-debloater, so I will try to package that instead.

**Comment by xfix** [comment]
Moved to a maintained fork.

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 269661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>universal-android-debloater</li>
  </ul>
</details>

File: pkgs/by-name/un/universal-android-debloater/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  '';
+
+  postInstall = ''
+    wrapProgram $out/bin/uad_gui \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ fontconfig freetype libglvnd xorg.libX11 xorg.libXcursor xorg.libXi xorg.libXrandr ]} \
+      --suffix PATH : ${lib.makeBinPath [ android-tools ]}
+  '';
+
+  meta = with lib; {
+    description = "A tool to debloat non-rooted Android devices";
```
Comment (WolfangAukang) [inline pkgs/by-name/un/universal-android-debloater/package.nix:55]:
```suggestion
    description = "A tool to debloat non-rooted Android devices";
    changelog = "https://github.com/Universal-Debloater-Alliance/universal-android-debloater/blob/${src.rev}/CHANGELOG.md";
```

**Review by WolfangAukang — COMMENTED** [review]
Just a small request. Rest LGTM

**Review by mdarocha — APPROVED** [review]

File: pkgs/by-name/un/universal-android-debloater/package.nix (hunk 1 above)
Comment (xfix) [reply pkgs/by-name/un/universal-android-debloater/package.nix:55]: Done.

**Review by WolfangAukang — APPROVED** [review]

File: pkgs/by-name/un/universal-android-debloater/package.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    changelog = "https://github.com/Universal-Debloater-Alliance/universal-android-debloater/blob/${src.rev}/CHANGELOG.md";
+    homepage = "https://github.com/Universal-Debloater-Alliance/universal-android-debloater";
+    license = licenses.gpl3;
```
Comment (wegank) [inline pkgs/by-name/un/universal-android-debloater/package.nix:58]:
```suggestion
    license = licenses.gpl3Only;
```

File: pkgs/by-name/un/universal-android-debloater/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    clang
+  ];
+
+  preCheck = ''
+    export HOME="$(mktemp -d)"
+  '';
+
+  postInstall = ''
+    wrapProgram $out/bin/uad_gui \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ fontconfig freetype libglvnd xorg.libX11 xorg.libXcursor xorg.libXi xorg.libXrandr ]} \
```
Comment (SuperSandro2000) [inline pkgs/by-name/un/universal-android-debloater/package.nix:50]: Can we avoid LD_LIBRARY_PATH somehow?

Comment (xfix) [reply pkgs/by-name/un/universal-android-debloater/package.nix:50]: This program depends on `iced_winit`, which depends on `winit`, which depends on `x11-dl`, which uses dlopen in https://github.com/AltF02/x11-rs/blob/ff57a942e5a4d35d8660362a8ca2ac0c02fa4852/x11-dl/src/link.rs. I have no idea how to patch Rust dependencies in Nixpkgs, so `LD_LIBRARY_PATH` is the simplest way around that.

Comment (xfix) [reply pkgs/by-name/un/universal-android-debloater/package.nix:50]: Yeah, looked into patching `winit` dependency, I don't think it's practical to change what https://github.com/rust-windowing/winit/blob/ca1674519ab3d8df4ce231fe018196a3981c7dea/src/platform_impl/linux/x11/xdisplay.rs does.

Comment (SuperSandro2000) [reply pkgs/by-name/un/universal-android-debloater/package.nix:50]: > Can we avoid LD_LIBRARY_PATH somehow?

This suggestion is crazy and will probably blow up lots of things started from that shell.

I think we can keep it for the wrapper if there is no good alternative.

**Review by wegank — APPROVED** [review]

**Pull request merged**